Source/objects.cpp

```diff
@@ -3974,26 +3974,11 @@ bool OperateFountains(int pnum, int i)
 
 void OperateWeaponRack(int pnum, int i, bool sendmsg)
 {
-	int weaponType;
-
 	if (Objects[i]._oSelFlag == 0)
 		return;
 	SetRndSeed(Objects[i]._oRndSeed);
 
-	switch (GenerateRnd(4) + ITYPE_SWORD) {
-	case ITYPE_SWORD:
-		weaponType = ITYPE_SWORD;
-		break;
-	case ITYPE_AXE:
-		weaponType = ITYPE_AXE;
-		break;
-	case ITYPE_BOW:
-		weaponType = ITYPE_BOW;
-		break;
-	case ITYPE_MACE:
-		weaponType = ITYPE_MACE;
-		break;
-	}
+	ItemType weaponType { PickRandomlyAmong({ ItemType::Sword, ItemType::Axe, ItemType::Bow, ItemType::Mace }) };

```

Comment:
This is a safe use of `PickRandomlyAmong` only because the list is a power of 2 in length.

Reply:
Can you elaborate why that's the case? The documentation states it 
> Generates a random integer less than the given limit

Why would it cause a problem if the list had non-power-of-2 items?

Reply:
Essentially this is due to the vanilla RNG codes dodgy implementation of Borland `rand()`. It tries to convert the current random engine state to a signed value using `abs()`. This fails when the engine state is UINT_MAX, the call to `abs()` returns INT_MIN (-2^31). `GenerateRnd()` then does modulo division on that value, if the limit/`v` is a power of 2 `-2^31 % v == 0` otherwise you end up with a negative number.

Reply:
There are a multitude of places in the codebase currently calling `GenerateRnd` with non-power-of-2 values... are _all_ of those bugs?

Because if they are, I don't really see the problem in this usage here.

Reply:
> are all of those bugs?

Yep 😢 .

I originally commented here to highlight that this was safe because we were lucky with the number of choices. However now that you've updated PickRandomlyAmong to always return an item in the list it is safe to use no matter how many elements are in the list.